Packs/EDL/Integrations/EDL/EDL.py

```diff
@@ -273,49 +593,56 @@ def is_valid_cidr(cidr: str) -> bool:
+def list_to_str(inp_list: list, delimiter: str = ',', map_func: Callable = str) -> str:
+    """
+    Transforms a list to an str, with a custom delimiter between each list item
     """
-    Create a list result of formatted_indicators
-     * Empty list:
-         1) if add_comment_if_empty, return {'# Empty EDL'}
+    str_res = ""
+    if inp_list:
+        if isinstance(inp_list, list):
+            str_res = delimiter.join(map(map_func, inp_list))
+        else:
+            raise AttributeError('Invalid inp_list provided to list_to_str')
+    return str_res
+
+
+def create_text_out_format(iocs: IO, request_args: RequestArguments) -> Union[IO, IO[str]]:
+    """
+    Create a list in new file of formatted_indicators
      * IP / CIDR:
          1) if collapse_ips, collapse IPs/CIDRs
      * URL:
-         1) if drop_invalids, drop invalids (length > 254 or has invalid chars)
+        1) if drop_invalids, drop invalids (length > 254 or has invalid chars)
+        2) if port_stripping, strip ports
+        3) if protocol_stripping, strip protocols
+        4) if url_truncate, truncate urls
     * Other indicator types:
         1) if drop_invalids, drop invalids (has invalid chars)
         2) if port_stripping, strip ports
     """
-    formatted_indicators = set()
     ipv4_formatted_indicators = set()
     ipv6_formatted_indicators = set()
-    for ioc in iocs:
+    iocs.seek(0)
+    formatted_indicators = tempfile.TemporaryFile(mode='w+t')
+    for str_ioc in iocs:
+        ioc = json.loads(str_ioc.rstrip())
         indicator = ioc.get('value')
         if not indicator:
             continue
         ioc_type = ioc.get('indicator_type')
-        # protocol stripping
-        indicator = _PROTOCOL_REMOVAL.sub('', indicator)
 
         if ioc_type not in [FeedIndicatorType.IP, FeedIndicatorType.IPv6,
                             FeedIndicatorType.CIDR, FeedIndicatorType.IPv6CIDR]:
-            # Port stripping
-            indicator_with_port = indicator
-            # remove port from indicator - from demisto.com:369/rest/of/path -> demisto.com/rest/of/path
-            indicator = _PORT_REMOVAL.sub(_URL_WITHOUT_PORT, indicator)
-            # check if removing the port changed something about the indicator
-            if indicator != indicator_with_port and not request_args.url_port_stripping:
-                # if port was in the indicator and url_port_stripping param not set - ignore the indicator
-                continue
-            # Reformatting to PAN-OS URL format
-            with_invalid_tokens_indicator = indicator
-            # mix of text and wildcard in domain field handling
-            indicator = _INVALID_TOKEN_REMOVAL.sub('*', indicator)
-            # check if the indicator held invalid tokens
+
+            indicator = url_handler(indicator, request_args.url_protocol_stripping,
+                                    request_args.url_port_stripping, request_args.url_truncate)
```

Comment:
Wasn't the `url_handler` already done as part of the `get_indicators_to_format` function?